modin/pandas/base.py

```diff
@@ -2894,7 +2894,7 @@ def value_counts(
     ):
         if subset is None:
             subset = self._query_compiler.columns
-        counted_values = self.groupby(by=subset, sort=False, dropna=dropna).size()
+        counted_values = self.groupby(by=subset, dropna=dropna, observed=True).size()
```

Comment:
1. `sort=False -> sort=True` so the users won't see the warning thrown by groupby
2. `observed=False -> observed=True` because newly added tests for `DataFrame.value_counts` showed that it's have to be set (see the example):

<details><summary>Example</summary>

```python
>>> df = pandas.DataFrame({"a": [1, 1, 2, 2], "b": [2, 2, 3, 3]}, dtype="category")
>>> df
   a  b
0  1  2
1  1  2
2  2  3
3  2  3
>>> df.value_counts()
a  b
1  2    2
2  3    2
dtype: int64
>>> df.groupby(["a", "b"]).size()
a  b
1  2    2
   3    0
2  2    0
   3    2
dtype: int64
>>> df.groupby(["a", "b"], observed=True).size()
a  b
1  2    2
2  3    2
dtype: int64
```

</details>